src/log/syslog.rs

```diff
@@ -1,90 +1,131 @@
 use core::fmt::{self, Write};
-use std::ffi::CStr;
-
 use log::{Level, Log, Metadata};
 
-use crate::system::syslog;
-
 pub struct Syslog;
 
-const LIMIT: usize = 960;
-const DOTDOTDOT_START: &[u8] = b"[...] ";
-const DOTDOTDOT_END: &[u8] = b" [...]";
-const NULL_BYTE: usize = 1; // for C string compatibility
-const BUFSZ: usize = LIMIT + DOTDOTDOT_END.len() + NULL_BYTE;
-const FACILITY: libc::c_int = libc::LOG_AUTH;
+mod internal {
+    use crate::system::syslog;
+    use std::ffi::CStr;
 
-struct SysLogWriter {
-    buffer: [u8; BUFSZ],
-    cursor: usize,
-    facility: libc::c_int,
-    priority: libc::c_int,
-}
+    const DOTDOTDOT_START: &[u8] = b"[...] ";
+    const DOTDOTDOT_END: &[u8] = b" [...]";
+
+    const MAX_MSG_LEN: usize = 960;
+    const NULL_BYTE_LEN: usize = 1; // for C string compatibility
+    const BUFSZ: usize = MAX_MSG_LEN + DOTDOTDOT_END.len() + NULL_BYTE_LEN;
 
-impl SysLogWriter {
-    fn new(priority: libc::c_int, facility: libc::c_int) -> Self {
-        Self {
-            buffer: [0; BUFSZ],
-            cursor: 0,
-            priority,
-            facility,
+    pub struct SysLogWriter {
+        buffer: [u8; BUFSZ],
+        cursor: usize,
+        facility: libc::c_int,
+        priority: libc::c_int,
+    }
+
+    // the caller of the pub functions will have to take care never to `append` more bytes than
+    // are `available`, or a panic will occur.
+    // the impl guarantees that after `line_break()`, there will be enough room available for at
+    // least a single UTF8 character sequence (which is true since MAX_MSG_LEN >= 10)
+    impl SysLogWriter {
+        pub fn new(priority: libc::c_int, facility: libc::c_int) -> Self {
+            Self {
+                buffer: [0; BUFSZ],
+                cursor: 0,
+                priority,
+                facility,
+            }
+        }
+
+        pub fn append(&mut self, bytes: &[u8]) {
+            let num_bytes = bytes.len();
+            self.buffer[self.cursor..self.cursor + num_bytes].copy_from_slice(bytes);
+            self.cursor += num_bytes;
+        }
+
+        pub fn line_break(&mut self) {
+            self.append(DOTDOTDOT_END);
+            self.commit_to_syslog();
+            self.append(DOTDOTDOT_START);
+        }
+
+        fn commit_to_syslog(&mut self) {
+            self.append(&[0]);
+            let message = CStr::from_bytes_with_nul(&self.buffer[..self.cursor]).unwrap();
+            syslog(self.priority, self.facility, message);
+            self.cursor = 0;
+        }
+
+        pub fn available(&self) -> usize {
+            MAX_MSG_LEN - self.cursor
         }
     }
 
-    fn append(&mut self, bytes: &[u8]) {
-        let num_bytes = bytes.len();
-        self.buffer[self.cursor..self.cursor + num_bytes].copy_from_slice(bytes);
-        self.cursor += num_bytes;
+    impl Drop for SysLogWriter {
```

Comment:
This does mean that you should always write a message at least once after creating a SysLogWriter, otherwise you will end up with an empty line in your syslog.

Reply:
That was also true of the previous code. I'll add a comment about that in the interface in concert with the renaming to "SysLogLineWriter".